saleor/graphql/core/utils/error_codes.py

```diff
@@ -0,0 +1,182 @@
+    ASSIGN_MENU_ITEM_TO_ITSELF = "assign_menu_item_to_itself"
+    CANNOT_ASSIGN_NOTE = "cannot_assign_node"
+    INVALID_MENU_ITEM = "invalid_menu_item"
+    NO_MENU_ITEM_PROVIDED = "no_item_provided"
+    TOO_MANY_MENU_ITEMS = "too_many_items"
+
+
+class OrderErrorCode(Enum):
+    CANNOT_CANCEL_FULFILLMENT = "cannot_cancel_fulfillment"
+    CANNOT_CANCEL_ORDER = "cannot_cancel_order"
+    CAPTURE_INACTIVE_PAYMENT = "capture_inactive_payment"
+    DELETE_LINE_NON_DRAFT_ORDER = "delete_line_non_draft_order"
+    DELETE_NON_DRAFT_ORDER = "delete_non_draft_order"
+    EDIT_NON_DRAFT_ORDER = "edit_non_draft_order"
+    FULFILL_ORDER_LINE = "fulfill_order_line"
+    ORDER_INVALID_SHIPPING_METHOD = "order_invalid_shipping_method"
+    ORDER_NO_PAYMENT = "order_no_payment"
+    ORDER_NO_PRODUCTS = "order_without_products"
+    ORDER_NO_SHIPPING_ADDRESS = "order_no_shipping_address"
+    QUANTITY_LESS_THAN_ONE = "quantity_less_than_one"
+    REFUND_MANUAL_PAYMENT = "refund_manual_payment"
+    VARIANT_DOES_NOT_EXIST = "variant_does_not_exist"
+    VOID_INACTIVE_PAYMENT = "void_inactive_payment"
+
+
+class CommonErrorCode(Enum):
+    AUTHORIZATION_KEY_ALREADY_EXISTS = "authorization_key_already_exists"
+    CANNOT_FETCH_TAX_RATES = "cannot_fetch_tax_rates"
+    DOES_NOT_EXIST = "does_not_exist"
+    GRAPHQL_ERROR = "graphql_error"
+    INCORRECT_VALUE = "incorrect_value"
+    INVALID_FILE_TYPE = "invalid_file_type"
+    INVALID_STOREFRONT_URL = "invalid_storefront_url"
+    INVALID_WEIGHT_UNIT = "invalid_weight_unit"
+    MISSING_VALUE = "missing_value"
+    NON_BLANK_VALUE_REQUIRED = "non_blank_value_required"
+    PARTIAL_PAYMENT_NOT_ALLOWED = "partial_payment_not_allowed"
+    PAYMENT_ERROR = "payment_error"
+    POSITIVE_NUMBER_REQUIRED = "positive_number_required"
+    SHIPPING_METHOD_REQUIRED = "shipping_method_required"
+    VALUE_ERROR = "value_error"
+
+
+class AttributeErrorCode(Enum):
+    ATTRIBUTE_ALREADY_ASSIGNED = "attribute_already_assigned"
+    ATTRIBUTE_ALREADY_EXISTS = "attribute_already_exists"
+    ATTRIBUTE_BAD_VALUE = "attribute_bad_value"
+    ATTRIBUTE_CANNOT_BE_ASSIGNED = "attribute_cannot_be_assigned"
+    ATTRIBUTE_DISABLED_VARIANTS = "attribute_disabled_variants"
+    ATTRIBUTE_NON_ASSIGNABLE = "attribute_non_assignable"
+    ATTRIBUTE_SLUG_ALREADY_EXISTS = "attribute_slug_already_exists"
+    ATTRIBUTE_SLUG_BLANK = "attribute_slug_blank"
+    ATTRIBUTE_VALUES_NOT_UNIQUE = "attribute_values_not_unique"
+
+
+class DigitalContentErrorCode(Enum):
+    MISSING_CONFIGURATION_FIELDS = "missing_configuration_fields"
+    VARIANT_NO_DIGITAL_CONTENT = "variant_no_digital_content"
+
+
+class ProductErrorCode(Enum):
+    NOT_PRODUCTS_IMAGE = "not_products_image"
+    PRODUCT_ALREADY_EXISTS = "product_already_exists"
+
+
+class ShippingErrorCode(Enum):
+    DEFAULT_SHIPPING_ZONE_ALREADY_EXISTS = "default_shipping_zone_already_exists"
+    MAX_LESS_THAN_MIN = "max_less_than_min"
+
+
+ERROR_CODE_UNKNOWN = "unknown"
```

Comment:
I'm not sure what I think about returning "unknown" code. First of all of shouldn't have such errors. Secondly, maybe instead we should just return `null` in that case?

Reply:
We should not, but something must be returned if some bad error code was used. 
For simplicity I've changed it to `null`.